packages/settingeditor/src/pluginlist.tsx

```diff
@@ -50,13 +50,21 @@ export class PluginList extends ReactWidget {
    */
   constructor(options: PluginList.IOptions) {
     super();
-    this.registry = options.registry;
+    this._registry = this.registry = options.registry;
     this.translator = options.translator || nullTranslator;
     this.addClass('jp-PluginList');
     this._confirm = options.confirm;
-    this.registry.pluginChanged.connect(() => {
+    this._model = options.model ?? new PluginList.Model(options);
+    this._model.ready
+      .then(() => {
+        this.update();
+      })
+      .catch(reason => {
+        console.error(`Failed to load the plugin list model:\n${reason}`);
+      });
+    this._model.changed.connect(() => {
```

Comment:
Should we connect this signal only after `this._model.ready` to ensure we don't trigger update when the model is not yet ready?